org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WChosenboxListEditor.java

```diff
@@ -622,4 +680,410 @@ public void newRecord(int record_ID) {
+							ListItem listItem = (ListItem) obj;
+							int index = selListbox.getIndexOfItem(listItem);
+							ListElement selObject = (ListElement) selModel.getElementAt(index);				
+							selObjects.add(selObject);						
+						}
+						migrateValueWithinSelectedList (selModel, selListbox, endIndex, selObjects);
+					}
+				}
+			}
+		}
+
+		private void migrateValueWithinSelectedList (SimpleListModel selModel, Listbox selListbox, int endIndex, List<ListElement> selObjects) {
+			int iniIndex =0;
+			Arrays.sort(selObjects.toArray());	
+			ListElement selObject= null;
+			ListElement endObject = (ListElement) selModel.getElementAt(endIndex);
+			for (ListElement selected : selObjects) {
+				iniIndex = selModel.indexOf(selected);
+				selObject = (ListElement) selModel.getElementAt(iniIndex);
+				selModel.removeElement(selObject);
+				endIndex = selModel.indexOf(endObject);
+				selModel.add(endIndex, selObject);			
+			}
+
+			selListbox.removeAllItems();
+			for(int i=0 ; i<selModel.getSize(); i++) { 	
+				ListElement pp = (ListElement) selModel.getElementAt(i);
+				selListbox.addItem(new ValueNamePair(pp.getID(), pp.getName()));
+			}
+		}
+
+		private void migrateValueWithinSelectedList (Event event) {
+			Object[] selObjects = selectedList.getSelectedItems().toArray();
+			if (selObjects == null)
+				return;
+			int length = selObjects.length;
+			if (length == 0)
+				return;
+			//
+			int[] indices = selectedList.getSelectedIndices();
+			//
+			boolean change = false;
+			//
+			Object source = event.getTarget();
+			if (source == bUp) {
+				for (int i = 0; i < length; i++) {
+					int index = indices[i];
+					if (index == 0)
+						break;
+					ListElement selObject = (ListElement) selectedModel.getElementAt(index);
+					ListElement newObject = (ListElement) selectedModel.getElementAt(index - 1);
+					selectedModel.setElementAt(newObject, index);
+					selectedModel.setElementAt(selObject, index - 1);
+					indices[i] = index - 1;
+					change = true;
+				}
+			}	//	up
+
+			else if (source == bDown) {
+				for (int i = length - 1; i >= 0; i--) {
+					int index = indices[i];
+					if (index  >= selectedModel.getSize() - 1)
+						break;
+					ListElement selObject = (ListElement) selectedModel.getElementAt(index);
+					ListElement newObject = (ListElement) selectedModel.getElementAt(index + 1);
+					selectedModel.setElementAt(newObject, index);
+					selectedModel.setElementAt(selObject, index + 1);
+					selectedList.setSelectedIndex(index + 1);
+					indices[i] = index + 1;
+					change = true;
+				}
+			}	//	down
+
+			//
+			if (change) {
+				selectedList.setSelectedIndices(indices);
+				if ( selectedList.getSelectedItem() != null) {
+					AuFocus focus = new AuFocus(selectedList.getSelectedItem());
+					Clients.response(focus);
+				}
+			}
+		}
+
+		private String getNewValue() {
+			return m_newValue;
+		}
+
+		private class ListElement extends NamePair {
```

Comment:
why do we need this instead of just using the existing ValueNamePair class ?

Reply:
we don't - is the result of useless copy-pastes